### PR TITLE
Retry the convertToSquahsfs request given the HPE_INVALID_CONSTANT error

### PR DESCRIPTION
**Pull request by Christian-Holbrook**
We occasionally see the convertToSquashfs request fail. It seems determined by the size of the zip and the length of the file name. The Roku responds with half of the response body and roku-deploy cannot handle the bad data.

The squashfs conversion looks like it successfully created, the error is in the response.

The work around is to make a second request to squashfs when we see the 'HPE_INVALID_CONSTANT' error. If that response includes 'fileType': 'squashfs' then we return the result saying it succeeded.

### CODE REVIEW

File: src/RokuDeploy.ts [hunk 1]
```diff
@@ -507,8 +507,23 @@ export class RokuDeploy {
             archive: '',
             mysubmit: 'Convert to squashfs'
         });
-
-        let results = await this.doPostRequest(requestOptions);
+        let results;
+        try {
+            results = await this.doPostRequest(requestOptions);
+        } catch (error) {
+            if ((error as any)?.code === 'HPE_INVALID_CONSTANT') {
```
Comment (TwitchBronBron) [inline src/RokuDeploy.ts:514]: We could use a comment here explaining what we're doing. Probably a more detailed comment would be best, cuz this is a weird one. :D

Comment (Christian-Holbrook) [reply src/RokuDeploy.ts:514]: 👍

Comment (Christian-Holbrook) [reply src/RokuDeploy.ts:518]: My comment is a bit wordy. Probably want to nitpick it a bit.

**Review by TwitchBronBron — APPROVED** [review]
